crates/api/src/v1/package.rs

```diff
@@ -10,27 +11,21 @@ use warg_protocol::{
     ProtoEnvelopeBody,
 };
 
-/// Represents the supported kinds of content sources.
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(tag = "type", rename_all = "camelCase")]
-pub enum ContentSource {
-    /// The content is located at an anonymous HTTP URL.
-    Http {
-        /// The URL of the content.
-        url: String,
-    },
-}
-
 /// Represents the supported kinds of content upload endpoints.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(tag = "type", rename_all = "camelCase")]
 pub enum UploadEndpoint {
-    /// Content may be uploaded via HTTP POST to the given URL.
-    /// If the endpoint responds with "201 Created" and a Location header, that
-    /// header's value will be the content source.
-    HttpPost {
+    /// Content may be uploaded via HTTP request to the given URL.
+    Http {
+        /// The http method for the upload request.
+        /// Only `POST` and `PUT` methods are accepted by the client.
+        method: String,
         /// The URL to POST content to.
         url: String,
+        /// Optional header names and values for the upload request.
+        /// Only `authorization` and `content-type` headers are accepted by the client.
```

Comment:
I think we should be stronger with this language. I think it's plausible that someone else would implement a partial client which would probably include this endpoint.

```suggestion
        /// Only `authorization` and `content-type` headers are valid; any other header should be rejected.
```